task_DEV_4/task_DEV_4/Parallelogram.cs

```diff
@@ -2,37 +2,56 @@
 
 namespace task_DEV_4
 {
-    class Parallelogram : Square
+    class Parallelogram : Figure
     {
-        public double HeightA { get; set; }
+        private const string MASSEGE_ERROR = "This value cannot be less than or equal to zero.";
         public double Angle { get; set; }
-        public double SideB { get; set; }
+        public double FirstSide { get; set; }        
+        public double SecondSide { get; set; }
 
-        public Parallelogram(double sideA, double heightA) : base(sideA)
+        public Parallelogram(string color, string name, double angle, double firstSide, double secondSide) : base(color, name)
         {
-            HeightA = heightA;
-        }
-
-        public Parallelogram(double sideA, double sideB, double angle) : base(sideA)
-        {
-            SideB = sideB;
             Angle = angle;
+            FirstSide = firstSide;
+            SecondSide = secondSide;
+            CheckForExistance();
         }
 
         public override double CalculateArea()
         {
-            if (HeightA > 0 && SideA > 0)
+            if (Angle > 0 || FirstSide > 0 || SecondSide > 0)
+            {
+                return FirstSide * SecondSide * Math.Sin(Angle);
+            }
+            else
             {
-                return SideA * HeightA;
+                throw new ArgumentOutOfRangeException(MASSEGE_ERROR);
             }
-            else if (SideA > 0 && SideB > 0 && Angle > 0)
+        }
+
+        public override double CalculatePerimeter()
+        {
+            if (FirstSide > 0 || SecondSide > 0)
```

Comment:
You dont need to do this check because you can be sure that your created figure is valid because you check it with CheckForExistance method when create.

Reply:
done